File: source-connector/src/main/java/sourceconnector/domain/log/EmptyLogMetadata.java
package sourceconnector.domain.log;

import offsetmanager.domain.file.FileKey;

public enum EmptyLogMetadata implements LogMetadata {
  INSTANCE;

  @Override
  public FileKey key() {
    // FIXME: return a proper empty FileKey if needed
    return null;
  }

  @Override
  public long offset() {
    return -1;
  }
}

File: source-connector/src/main/java/sourceconnector/domain/log/EmptyLogMetadata.kt
package sourceconnector.domain.log

import offsetmanager.domain.file.FileKey

enum class EmptyLogMetadata(
  private val emptyFileKey: FileKey = EmptyFileKey(),
  private val emptyOffset: Long = -1
) : LogMetadata {
  INSTANCE;

  override fun key(): FileKey {
    return emptyFileKey
  }

  override fun offset(): Long {
    return emptyOffset
  }
}
